### PR TITLE
internal: Add Plugin client

### PR DESCRIPTION
**Pull request by abhinav**
This adds a cleaner client API on top of generated Thrift APIs. This will be
used by the code generator to talk to plugins.

CC @breerly @prashantv @kriskowal

### CODE REVIEW

**Comment by prashantv** [comment]
lgtm

**Comment by breerly** [comment]
:shipit: (nice tests sir)

File: internal/plugin/transport.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	"github.com/thriftrw/thriftrw-go/internal/multiplex"
+	"github.com/thriftrw/thriftrw-go/plugin/api"
+	"github.com/thriftrw/thriftrw-go/plugin/api/service/plugin"
+	"github.com/thriftrw/thriftrw-go/plugin/api/service/servicegenerator"
+	"github.com/thriftrw/thriftrw-go/protocol"
+
+	"github.com/uber-go/atomic"
+)
+
+var _proto = protocol.Binary
```
Comment (kriskowal) [inline internal/plugin/transport.go]: What’s the idea behind this underscore?

**Comment by kriskowal** [comment]
lgtm

File: internal/plugin/transport.go (hunk 1 above)
Comment (abhinav) [inline internal/plugin/transport.go]: Just a convention for variables that should be treated as local to the current file rather than the package.

**Pull request merged**